From 388a91b06473b13a59ba217d23b26ca3dc0f2555 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Fri, 7 Aug 2015 00:06:15 -0400
Subject: [PATCH] man: clarify that unknown escapes must be escaped

https://bugzilla.redhat.com/show_bug.cgi?id=1251334
is about a unit file which has
Environment=TERM=linux PS1=system-upgrade:\w\$\x20
We used to allow that, but after recent tightening of parsing
rules, we barf. Make it clear that this is intentional.
---
 man/systemd.service.xml | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/man/systemd.service.xml b/man/systemd.service.xml
index 4368ca8..7e96989 100644
--- a/man/systemd.service.xml
+++ b/man/systemd.service.xml
@@ -922,7 +922,10 @@
     the arguments. Double quotes ("...") and single quotes ('...') may
     be used, in which case everything until the next matching quote
     becomes part of the same argument. C-style escapes are also
-    supported, see table below. Quotes themselves are removed after
+    supported. The table below contains the list of allowed escape
+    patterns. Only patterns which match the syntax in the table are
+    allowed; others will result in an error, and must be escaped by
+    doubling the backslash. Quotes themselves are removed after
     parsing and escape sequences substituted. In addition, a trailing
     backslash (<literal>\</literal>) may be used to merge lines.
     </para>
@@ -939,7 +942,7 @@
     <literal>&amp;</literal>, and <emphasis>other elements of shell
     syntax are not supported</emphasis>.</para>
 
-    <para>The command to execute must an absolute path name. It may
+    <para>The command to execute must be an absolute path name. It may
     contain spaces, but control characters are not allowed.</para>
 
     <para>The command line accepts <literal>%</literal> specifiers as
-- 
2.4.3

